### PR TITLE
[READY] Better explosion echoes

### PR DESCRIPTION
**Pull request by NotRanged**
## About The Pull Request

Modifies explosion code to play the distant sounds significantly futher. Remote explosions that have a heavy range are siginificant enough to echo through the hull no matter where you are onboard.

## Why It's Good For The Game
Testing it locally, this made events like a loose tesla and a meteor storm truly terrifying throughout the station. It makes sense for siginificant bangs to be heard through the hull of the station anyway!

Clip of remote explosion: https://streamable.com/6xap9j

Clip of a tesloose: https://streamable.com/yel0od

Now all of engineering will know when the armory blows up.

## Changelog
:cl:
tweak: Changed explosion sound code to be more noticeable and terrifying throughout the station during siginificant explosive events.
/:cl:

### CODE REVIEW

File: code/datums/explosion.dm [hunk 1]
```diff
@@ -103,13 +103,14 @@ GLOBAL_LIST_EMPTY(explosions)
 	// 3/7/14 will calculate to 80 + 35
 
 	var/far_dist = 0
-	far_dist += heavy_impact_range * 5
+	far_dist += heavy_impact_range * 15
```
Comment (jjpark-kb) [inline code/datums/explosion.dm:106]: //skyrat edit

File: code/datums/explosion.dm [hunk 2]
```diff
@@ -103,13 +103,14 @@ GLOBAL_LIST_EMPTY(explosions)
...
 	far_dist += devastation_range * 20
 
 	if(!silent)
 		var/frequency = get_rand_frequency()
 		var/sound/explosion_sound = sound(get_sfx("explosion"))
 		var/sound/far_explosion_sound = sound('sound/effects/explosionfar.ogg')
+		var/sound/explosion_echo_sound = sound('sound/effects/explosionecho.ogg')
```
Comment (jjpark-kb) [inline code/datums/explosion.dm:113]: //skyrat edit

File: code/datums/explosion.dm [hunk 3]
```diff
@@ -126,11 +127,18 @@ GLOBAL_LIST_EMPTY(explosions)
...
+						M.playsound_local(epicenter, null, far_volume, 1, frequency, S = far_explosion_sound, distance_multiplier = 0) // Far sound
+					else
+						M.playsound_local(epicenter, null, far_volume, 1, frequency, S = explosion_echo_sound, distance_multiplier = 0) // Echo sound
 					if(baseshakeamount > 0)
 						shake_camera(M, 10, clamp(baseshakeamount*0.25, 0, 2.5))
+				else if(M.can_hear() && !isspaceturf(get_turf(M)) && heavy_impact_range) // Big enough explosions echo throughout the hull
+					var/echo_volume = 30
+					if(devastation_range)
+						echo_volume = 50
+					M.playsound_local(epicenter, null, echo_volume, 1, frequency, S = explosion_echo_sound, distance_multiplier = 0)
```
Comment (jjpark-kb) [inline code/datums/explosion.dm:141]: //skyrat edit

**Comment by ShadeAware** [comment]
Little bit of a suggestion, why not also add sounds for the station structure creaking with explosions? Theres a lot of good ones on the NSV repo that would work well.

**Comment by NotRanged** [comment]
> Little bit of a suggestion, why not also add sounds for the station structure creaking with explosions? Theres a lot of good ones on the NSV repo that would work well.

I don't play on NSV, and I can't seem to find the sound files you're talking about. If you send them I could add them.

**Comment by NotRanged** [comment]
All done :) Please testmerge or fullmerge.

**Comment by ShadeAware** [comment]
Y E S

**Pull request merged**